token/client/src/token.rs

```diff
@@ -2206,57 +2213,42 @@ where
         .await
     }
 
-    /// Applies the confidential transfer pending balance to the available balance using the
-    /// uniquely derived decryption key
-    #[cfg(feature = "proof-program")]
-    pub async fn confidential_transfer_apply_pending_balance<S: Signer>(
+    /// Applies the confidential transfer pending balance to the available balance
+    pub async fn confidential_transfer_apply_pending_balance<S: Signers>(
         &self,
-        token_account: &Pubkey,
-        authority: &S,
-        available_balance: u64,
-        pending_balance: u64,
-        expected_pending_balance_credit_counter: u64,
+        account: &Pubkey,
+        authority: &Pubkey,
+        account_info: Option<ApplyPendingBalanceExtensionInfo>,
```

Comment:
This can take the extension directly if you add the functions on the extension struct type rather than using a new type

Reply:
Looking at it now, implementing the functions directly on `ConfidentialTransferAccount` could be quite clean if we are to refactor the `extension_info` module to token-2022.

I think the main benefit of having a separate struct `ApplyPendingBalanceExtensionInfo` (and for the client function to take in this struct) would be that it holds just the fields from a `ConfidentialTransferAccount` that are needed to generate an `ApplyPendingBalance` instruction and hence it is much shorter than `ConfidentialTransferAccount`. 

I thought this could be useful for something like offline signing in the cli. Users can first look-up the relevant data from a confidential transfer account encoded as `ApplyPendingBalanceExtensionInfo`, copy it, and then use it as input to sign a transaction offline. In this case, a shorter `ApplyPendingBalanceExtensionInfo` would be more convenient than copying the entire `ConfidentialTransferAccount`.

What do you think about this aspect?

Reply:
I see what you mean -- it makes sense to encapsulate this in a separate type for clean separation. How many of these types will you need to support the rest of the instructions? Will it be one per instruction?

Reply:
Yeah, so I think we will need it for the instructions that need zkp's (`EmiptyAccount`, `Withdraw`, `Transfer`, `TransferWithFee` and `WithdrawWithheldTokens`) + `ApplyPendingBalance`.